vpn/tunnel.go

```diff
@@ -130,6 +153,24 @@ func (t *Tunnel) handleRPC(req *ManagerMessage, msgID uint64) *TunnelMessage {
 	}
 }
 
+func UseAsRouter() TunnelOption {
+	return func(t *Tunnel) {
+		t.router = NewRouter(t)
+	}
+}
+
+func UseAsLogger() TunnelOption {
+	return func(t *Tunnel) {
+		t.logger = t.logger.AppendSinks(t)
+	}
+}
+
+func UseAsDNSConfig() TunnelOption {
+	return func(t *Tunnel) {
+		t.dnsConfigurator = NewDNSConfigurator(t)
+	}
+}
```

Comment:
Do we ever not want these things? They should just be the default and we should remove `TunnelOption` until the need arises

Reply:
Spike suggested earlier: https://github.com/coder/coder/pull/15612#discussion_r1853346826

Reply:
I honestly prefer the separate build tag files that do the implementations for each of these on each platform but to each their own